Assets/MixedRealityToolkit/Utilities/Editor/Setup/MixedRealityToolkitFiles.cs

```diff
@@ -29,42 +43,82 @@ public static void OnPostprocessAllAssets(string[] importedAssets, string[] dele
                 foreach (string asset in importedAssets.Concat(movedAssets))
                 {
                     string folder = asset.Replace("Assets", Application.dataPath);
-                    if (folder.EndsWith(MixedRealityToolkitDirectory))
+                    foreach (MixedRealityToolkitModuleType module in Enum.GetValues(typeof(MixedRealityToolkitModuleType)))
                     {
-                        mrtkFolders.Add(NormalizeSeparators(folder));
+                        if (folder.EndsWith(MixedRealityToolkitDirectory(module)))
+                        {
+                            if (!mrtkFolders.TryGetValue(module, out HashSet<string> modFolders))
+                            {
+                                modFolders = new HashSet<string>();
+                                mrtkFolders.Add(module, modFolders);
+                            }
+                            modFolders.Add(NormalizeSeparators(folder));
+                        }
                     }
                 }
 
                 foreach (string asset in deletedAssets.Concat(movedFromAssetPaths))
                 {
                     string folder = asset.Replace("Assets", Application.dataPath);
-                    if (folder.EndsWith(MixedRealityToolkitDirectory))
+                    foreach (MixedRealityToolkitModuleType module in Enum.GetValues(typeof(MixedRealityToolkitModuleType)))
                     {
-                        folder = NormalizeSeparators(folder);
-                        if (mrtkFolders.Contains(folder) && !Directory.Exists(folder))
+                        if (mrtkFolders.TryGetValue(module, out HashSet<string> modFolders))
```

Comment:
>mrtkFolders [](start = 28, length = 11)

Since we're now storing a Dictionary of HashSets, it's possible for the underlying hashsets to end up being empty after this operation - in which case, it's probably right to also delete the entry from the dictionary. 

Otherwise you'll get some functions below returning incorrect information like:

public static bool AreFoldersAvailable
        {
            get
            {
                searchForFoldersTask.Wait();
                return mrtkFolders.Count > 0;
            }
        }

Which will check for the presence of data in the dictionary (which won't be correct) instead of what they want to do (which is check that there is a non-empty hashset contained within that dictionary).

Anyways, just doing a size == 0, then remove, would fix these issues.